src/OpenSpaceToolkit/Astrodynamics/Trajectory/Segment.cpp

```diff
@@ -663,42 +830,36 @@ Segment::Solution Segment::Solve_(
         states.add(stateBuilder.expand(state.inFrame(aState.accessFrame()), aState));
     }
 
-    return {
-        name_,
-        aDynamicsArray,
-        states,
-        conditionSolution.conditionIsSatisfied,
-        type_,
-    };
-}
-
-const Shared<Thruster> Segment::FindThrusterDynamics(const Array<Shared<Dynamics>>& aDynamicsArray)
-{
-    // Loop through dynamics to find Thruster dynamics
-    Shared<Thruster> thrusterDynamics = nullptr;
-    bool thrusterDynamicsFound = false;
-    for (const Shared<Dynamics>& dynamic : aDynamicsArray)
+    // Determine the final conditionIsSatisfied value
+    bool finalConditionIsSatisfied = conditionSolution.conditionIsSatisfied;
```

Comment:
I'm not sure I understand why you may need the condition re-evaluated? Can you leave a comment explaining what situation this happens in?

Reply:
Added a comment.